cli/cmd/install.go

```diff
@@ -320,20 +332,61 @@ func render(w io.Writer, values *l5dcharts.Values, stage string) error {
 		}
 	}
 
-	// TODO refactor to use l5dcharts.LoadChart()
-	chart := &charts.Chart{
-		Name:      helmDefaultChartName,
-		Dir:       helmDefaultChartDir,
-		Namespace: controlPlaneNamespace,
-		RawValues: rawValues,
-		Files:     files,
-		Fs:        static.Templates,
+	var partialFiles []*loader.BufferedFile
+	for _, template := range charts.L5dPartials {
+		partialFiles = append(partialFiles,
+			&loader.BufferedFile{Name: template},
+		)
+	}
+
+	// Load all Viz chart files into buffer
+	if err := charts.FilesReader(static.Templates, helmDefaultChartDir+"/", files); err != nil {
+		return err
+	}
+
+	// Load all partial chart files into buffer
+	if err := charts.FilesReader(static.Templates, "", partialFiles); err != nil {
+		return err
+	}
+
+	// Create a Chart obj from the files
+	chart, err := loader.LoadFiles(append(files, partialFiles...))
+	if err != nil {
+		return err
+	}
+
+	// Store final Values generated from values.yaml and CLI flags
+	err = yaml.Unmarshal(rawValues, &chart.Values)
+	if err != nil {
+		return err
+	}
+
+	// Create values override
+	valuesOverrides, err := options.MergeValues(nil)
+	if err != nil {
+		return err
+	}
+
+	vals, err := chartutil.CoalesceValues(chart, valuesOverrides)
```

Comment:
If I'm reading this correctly, this matches what I describe in https://github.com/linkerd/linkerd2/issues/5158#issuecomment-758306503.  i.e. `MergeValues` will return a set of value overrides generated from the `--set` and `--values` flags and then these overrides will be applied on top of the values loaded from the the Linkerd cluster.

Reply:
Yes, That is right as `chart.Values` contains the (default+current cluster config+CLI custom flags) while `vals` is through helm customization flags.